nemo_skills/inference/reward_model.py

```diff
@@ -0,0 +1,175 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import importlib
+import json
+import logging
+import sys
+from copy import deepcopy
+from dataclasses import asdict, field
+from pathlib import Path
+
+import hydra
+from tqdm import tqdm
+
+from nemo_skills.inference.server.code_execution_model import ErrorRecoveryConfig, server_params
+from nemo_skills.inference.server.reward_model import get_reward_model
+from nemo_skills.prompt.utils import get_prompt
+from nemo_skills.utils import get_fields_docstring, get_help_message, nested_dataclass, setup_logging
+
+LOG = logging.getLogger(__file__)
+
+
+@nested_dataclass(kw_only=True)
+class RewardModelConfig:
+    """LLM reward model parameters."""
+
+    output_file: str | None = None  # Where to save the generations
+    # Inference server configuration {server_params}
+    server: dict = field(default_factory=dict)
+    # Prompt configuration - path to yaml files
+    prompt_template: str | None = None  # not required for OpenAI server
+    prompt_config: str  # we will fetch it from dataset dir if not provided
+
+    input_file: str | None = None  # Can directly specify an input file, if using a custom dataset
+    input_dir: str | None = None  # Can specify an input direct
```

Comment:
Can you please add a short comment on how input_dir/random_seed are used? Especially for random seed as it's easy to confuse it with generation parameter

Reply:
added in 5337266e7a2e0a7cb3d252b66bb8f0f2b546e753